lib/workflows/create_jasmin_external_project.py

```diff
@@ -0,0 +1,180 @@
+from typing import Optional, List
+
+from enums.network_providers import NetworkProviders
+from enums.rbac_network_actions import RbacNetworkActions
+from enums.user_domains import UserDomains
+
+from openstack.connection import Connection
+
+from openstack_api.openstack_network import (
+    create_network,
+    create_network_rbac,
+    allocate_floating_ips,
+)
+from openstack_api.openstack_project import create_project
+from openstack_api.openstack_quota import set_quota
+from openstack_api.openstack_router import create_router, add_interface_to_router
+from openstack_api.openstack_security_groups import (
+    refresh_security_groups,
+    create_https_security_group,
+    create_http_security_group,
+    create_external_security_group_rules,
+)
+from openstack_api.openstack_roles import assign_role_to_user
+from openstack_api.openstack_subnet import create_subnet
+from structs.network_details import NetworkDetails
+from structs.network_rbac import NetworkRbac
+
+from structs.project_details import ProjectDetails
+from structs.quota_details import QuotaDetails
+from structs.role_details import RoleDetails
+from structs.router_details import RouterDetails
+
+
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-locals
+def create_jasmin_external_project(
+    conn: Connection,
+    project_name: str,
+    project_email: str,
+    domain: str,
+    project_description: str,
+    network_name: str,
+    subnet_name: str,
+    router_name: str,
+    number_of_floating_ips: int = 1,
+    number_of_security_group_rules: int = 200,
```

Comment:
both of these need `Optional[int]` type hinting

Reply:
This section isn't part of getting the initial workflow up - to be done in a separate PR (or built on this PR)

Reply:
TBH I think we should drop these args across the board and hardcode them to whatever ops say rather than passing them as args everywhere
Instead we should make quota changes using the dedicated action later on and start with sensible defaults

It's out of scope for this PR but I'd like to do a follow-on PR that nukes this everywhere